src/metadata.rs

```diff
@@ -2148,6 +2148,100 @@ mod test {
         assert_eq!(decoded, root);
     }
 
+    fn jsn_root_metadata_without_keyid_hash_algos() -> serde_json::Value {
+        json!({
+            "_type": "root",
+            "spec_version": "1.0",
+            "version": 1,
+            "expires": "2017-01-01T00:00:00Z",
+            "consistent_snapshot": false,
+            "keys": {
+                "09557ed63f91b5b95917d46f66c63ea79bdaef1b008ba823808bca849f1d18a1": {
```

Comment:
I think this keyid is wrong. If you look above in `serde_root_metadata`, this keyid matches the one that includes `keyid_hash_algorithms`.

Reply:
Updated. Should we verify the key ids when we deserialize?